React/CSS-in-JS/Chakra.md

```diff
@@ -0,0 +1,629 @@
+
+<!-- TODO maybe what should be defined in global -->
+
+Example:
+
+```ts
+const overrides = {
+  styles: {
+    global: {
+      body: {
+        fontFamily: "body",
+        bg: "white",
+        color: "primary.200",
+      },
+      "*": {
+        boxSizing: "border-box",
+      },
+    },
+  },
+};
+```
+
+### Responsive
+
+Chakra UI supports responsive styles out of the box. Instead of manually adding @media queries and adding nested styles throughout your code, Chakra UI allows you to provide object and array values to add mobile-first responsive styles.
+
+> Under the hood `@media(min-width)` media query is used to ensure mobile-first
+
+Chakra UI default breakpoints:
+
+```ts
+export const breakpoints = {
+  sm: "30em",
+  md: "48em",
+  lg: "62em",
+  xl: "80em",
+};
+```
+
+Here's how to interpret this syntax:
+
+- `base`: From `0em` upwards
+- `md`: From `48em` upwards
+- `lg`: From `62em` upwards
+- `xl`: From `80em` upwards
+
+For responsive styles, array or object syntax can be used.
+
+```tsx
+// array syntax
+<Text fontSize={["24px", "40px", "56px"]}>
+  Lorem Ipsum is simply dummy text
+</Text>
+```
+
+```tsx
+// object syntax
+<Text fontSize={{ base: "24px", md: "40px", lg: "56px" }}>
+  Lorem Ipsum is simply dummy text
+</Text>
+```
+
+In case we need to skip cretan breakpoint, `null` is passed at that position in the array to avoid generating unnecessary CSS.
+
+Example:
+
+```jsx
+<Text fontSize={["24px", null, "56px"]}>Lorem Ipsum is simply dummy text</Text>
+```
+
+Array and Object syntax work for every style prop in the theme specification, which means you can change most properties at a given breakpoint.
+
+To create custom breakpoints read [docs](https://chakra-ui.com/docs/features/responsive-styles#customizing-breakpoints).
+
+For more about responsive styles read [styled-system responsive styles docs](https://styled-system.com/responsive-styles/)
+
+### Chakra Factory
+
+Chakra factory serves as an object of chakra JSX elements, and also a function that can be used to enable custom component receive chakra's style props.
+
+```tsx
+import { chakra } from "@chakra-ui/react";
+
+// object
+<chakra.button
+  sx={{
+    shadow: "lg",
+    rounded: "lg",
+    bg: "white",
+  }}
+>
+  Click me
+</chakra.button>;
+
+// function
+chakra("button", {
+  baseStyle: {
+    shadow: "lg",
+    rounded: "lg",
+    bg: "white",
+  },
+});
+```
+
+This reduces the need to create custom component wrappers and name them. Syntax is available for common html elements. See the reference for the full [list of elements](https://github.com/chakra-ui/chakra-ui/blob/develop/packages/system/src/system.utils.ts#L9) supported.
+
+For example `react-datepicker` can be wrapped in chakra factory function so we can pass `sx` or style props to `<DatePicker />`
+
+Example:
+
+```tsx
+import DatePicker from "react-datepicker";
+import { chakra } from "@chakra-ui/react";
+
+const ChakraDatepicker = chakra(DatePicker);
```

Comment:
Can we use `Styled` prefix... Chakra use it in the source code... so
const StyledaDatepicker = chakra(DatePicker);

Reply:
For example here https://github.com/chakra-ui/chakra-ui/blob/develop/packages/menu/src/menu.tsx#L71